content/blog/2021-12-07-ml-experiment-versioning.md

```diff
@@ -0,0 +1,293 @@
+
+# Distributed Experiments
+
+Experiment tracking tools log experiments to a central database and show them in
+a dashboard. This enables comparison that would be difficult in Git. Distributed
+experiments are recorded in the local code repo like leaves or twigs growing
+from Git branches.
+
+```dvc
+$ dvc exp show
+┏━━━━━━━━━━━━━━━━━━━━━━━━━┳━━━━━━━━━━━━━━┳━━━━━━━━━┳━━━━━━━━┳━━━━━━━━━━━━━━┳━━━━━━━━━━━━━━━━━━┓
+┃ Experiment              ┃ Created      ┃    loss ┃    acc ┃ train.epochs ┃ model.conv_units ┃
+┡━━━━━━━━━━━━━━━━━━━━━━━━━╇━━━━━━━━━━━━━━╇━━━━━━━━━╇━━━━━━━━╇━━━━━━━━━━━━━━╇━━━━━━━━━━━━━━━━━━┩
+│ workspace               │ -            │ 0.25183 │ 0.9137 │ 10           │ 64
+│
+│ mybranch                │ Oct 23, 2021 │       - │      - │ 10           │ 16
+│
+│ ├── 9a4ff1c [exp-333c9] │ 10:40 AM     │ 0.25183 │ 0.9137 │ 10           │ 64
+│
+│ ├── 138e6ea [exp-55e90] │ 10:28 AM     │ 0.25784 │ 0.9084 │ 10           │ 32
+│
+│ └── 51b0324 [exp-2b728] │ 10:17 AM     │ 0.25829 │ 0.9058 │ 10           │ 16
+│
+└─────────────────────────┴──────────────┴─────────┴────────┴──────────────┴──────────────────┘
+```
+
+You may push and pull what you want to share. When collaborating, only a small
+selection of experiments may be worth sharing, and you may not know until after
+you run and compare them. Pushing an experiment shares all the experiment
+information and code so that they stay in sync and are shared with the same
+people, and it's already organized using Git branches.
+
+```dvc
+$ dvc exp push origin exp-333c9
+Pushed experiment 'exp-333c9'to Git remote 'origin'.
+```
+
+While Git branching is a flexible way to organize experiments, there are often
+too many experiments to fit any Git branching workflow. Experiments must act
+like leaves that may be blown away easily, yet also like twigs that may grow
+into branches:
+
+```dvc
+$ dvc exp branch exp-333c9 conv-units-64
+Git branch 'conv-units-64' has been created from experiment 'exp-333c9'.
+To switch to the new branch run:
+
+        git checkout conv-units-64
+
+```
+
+# What Next?
+
+There are parallels between the
+[history of version control](https://ericsink.com/vcbe/html/history_of_version_control.html)
+and the history of experiment tracking. Git's distributed nature and incremental
+change tracking were major advances over the centralized, file-based version
+control systems of previous generations. Experiment versioning can similarly
+advance the next generation of experiment tracking.
+
+Experiment versioning is still in its early days. Look out for future
+announcements about:
+
+- Deep learning features like [live monitoring](https://dvc.org/doc/dvclive) and
+  [checkpointing](https://dvc.org/doc/user-guide/experiment-management/checkpoints).
+- Visualizing and comparing experiment results in other tools like VS Code and
+  [DVC Studio](https://studio.iterative.ai/).
+- Orchestrating experiments on other
+  [machines](https://github.com/iterative/dvc/wiki/Remote-executors).
```

Comment:
It might be a bit too early for `dvc machines` 😄 

Should we point to our Twitter or YouTube?

Reply:
> Should we point to our Twitter or YouTube?

@jendefig What do you think?